src/flakeref.rs

```diff
@@ -408,25 +398,19 @@ impl FlakeRefType {
                     let owner = parsed_iter.next().ok_or(er_fn("owner"))?;
                     let repo = parsed_iter.next().ok_or(er_fn("repo"))?;
                     let ref_or_rev = parsed_iter.next();
-                    let flake_ref_type = match flake_ref_type_str {
-                        "github" => FlakeRefType::GitHub {
-                            owner,
-                            repo,
-                            ref_or_rev,
-                        },
-                        "gitlab" => FlakeRefType::GitLab {
-                            owner,
-                            repo,
-                            ref_or_rev,
-                        },
-                        "sourcehut" => FlakeRefType::Sourcehut {
-                            owner,
-                            repo,
-                            ref_or_rev,
-                        },
+                    let src = match flake_ref_type_str {
+                        "github" => GitForge::GitHub,
+                        "gitlab" => GitForge::GitLab,
+                        "sourcehut" => GitForge::SourceHut,
```

Comment:
I feel this should be rolled up into a from_str trait impl. we could do something like:
```rust
if let Ok(some_gitforge) = ... {
    ...
} else {
    // match on the other possible strings
}
```

...in order to get rid of the unreachable!, but that's probably getting a touch fancy.